src/views/common/Animation/AnimationUtilities/useTimeSyncing.ts

```diff
@@ -0,0 +1,122 @@
+    refs.targetExternalTimeRef.current = state.currentTime
+    return true
+}
+const timeResolver: DebounceThrottleResolver<debounceUpdateRefs, debounceUpdateResolverProps> = (refs, props) => {
+    // It looks like this is actually getting called too often--not sure what's going on there.
+    // console.log(`Updating time, it is ${Date.now()}`)
+    props.committer(refs.targetExternalTimeRef.current, props.setterFn)
+}
+
+const checkNewTimeIsWithinEpsilon = (scheduledNewTimeRef: React.MutableRefObject<number | undefined>, frameTime: number | undefined, replayRateMultiplier: number) => {
+    if (scheduledNewTimeRef.current === undefined || frameTime === undefined) return false
+    // At 1:1 playback, we expect ~100 ms to pass between 100-ms-throttled focus-update calls. Just need to multiply by replay rate multiplier.
+    const epsilon = Math.abs(throttleRateMs * replayRateMultiplier * 1.5 * .001) // the 1.5 multiplier gives us a bit of fudge, while .001 converts MS to S.
+    return Math.abs(scheduledNewTimeRef.current - frameTime) < epsilon
+}
+
+const useLiveTimeSyncing = <T, >(setExternalTime: timeSetter, state: AnimationState<T>, dispatch: React.Dispatch<AnimationStateAction<T>>, getTimeFromFrame: (frame: T) => number) => {
+    const targetExternalTimeRef = useRef<number | undefined>(undefined)
+    const refs: debounceUpdateRefs = useMemo(() => { return { targetExternalTimeRef }}, [targetExternalTimeRef])
+    const findNearestTime = useTimeLookupFn(state, getTimeFromFrame)
+    const resolverProps = useMemo(() => { return { committer: syncOutsideTimeToFrameTime, setterFn: setExternalTime }  }, [setExternalTime])
+    const { throttler: throttledSyncOutsideTimeToFrameTime, cancelThrottled } = useThrottler(timeUpdater, timeResolver, refs, resolverProps, throttleRateMs)
+
+    const epsilonCheck = useCallback(
+        (frameTime: number | undefined, replayRateMultiplier: number) => checkNewTimeIsWithinEpsilon(refs.targetExternalTimeRef, frameTime, replayRateMultiplier),
+        [refs])
+
+    const handleOutsideTimeUpdate = useCallback((otherTime: number | undefined) => liveSyncFrameToOutsideTime({
+        outsideTime: otherTime,
+        checkNewTimeWithinEpsilon: epsilonCheck,
+        multiplier: state.replayMultiplier,
+        cancelPendingUpdateOfOutsideTime: cancelThrottled,
+        findNearestTime,
+        dispatch
+    }), [epsilonCheck, state.replayMultiplier, cancelThrottled, findNearestTime, dispatch])
+
+    const handleFrameTimeUpdate = useCallback(() => {
+        const internalTime = getTimeFromFrame(state?.frameData[state?.currentFrameIndex])
+        state.isPlaying ? throttledSyncOutsideTimeToFrameTime({currentTime: internalTime}) : syncOutsideTimeToFrameTime(internalTime, setExternalTime)
+    }, [getTimeFromFrame, state.isPlaying, state.frameData, state.currentFrameIndex, throttledSyncOutsideTimeToFrameTime, setExternalTime])
+
+    return { handleOutsideTimeUpdate, handleFrameTimeUpdate }
```

Comment:
At the end of the day, all of this logic exposes two functions which should be called in `useEffect` hooks in the consumer. One of them should be called every time the external time (i.e. focusTime) changes: that's the one that syncs playback frame to the external time frame. During playback this will do nothing if the focus time and the playback time are close enough--in order to account for the throttled callback updating the focus time from the playback time--but clicking to update the focus still works during playback for realistically far-away values.

The other function can be run whenever playback is started or stopped, and just syncs the focus state to the frame state.